sdk/python/feast/permissions/auth/oidc_token_parser.py

```diff
@@ -88,19 +88,16 @@ async def user_details_from_access_token(self, access_token: str) -> User:
             current_user = data["preferred_username"]
 
             if "resource_access" not in data:
-                raise AuthenticationError(
```

Comment:
Are we sure? shouldn't we expose an authorization error instead? 
the token does not include any relevant information so, we should highlight this fact

Reply:
I'm not sure what `resource_access` contains. The problem is that an unauthorized user is getting an authentication error, but the issue is just that the user doesn't have any roles assigned. Ideally, the error message should be more like what Kubernetes examples gives in this situation related with permissions/access.

Reply:
This is an example of `resource_access` in the access token from an `app` client:
```json
  "resource_access": {
    "app": {
      "roles": [
        "reader"
      ]
    }
```
If you want to remove these errors please at least use `logger.warning` instead of `logger.info`.

BTW: the `KubernetesTokenParser` does not raise any explicit error if no roles are found, but if the lookup of K8s resources fail, the k8s client library can raise errors that would be forwarded to the caller and then result in some `Internal Error` or similar. Also, the `_decode_token` method of that class raise an `AuthenticationError` if the decoded token doe not include the `sub` field.

```py
        role_bindings = self.rbac_v1.list_namespaced_role_binding(namespace)
        cluster_role_bindings = self.rbac_v1.list_cluster_role_binding()
```